pkg/providers/docker/docker.go

```diff
@@ -52,6 +58,14 @@ type provider struct {
 	templateBuilder       *DockerTemplateBuilder
 }
 
+// KubeconfigReader reads the kubeconfig secret on a cluster and copies the contents to a writer.
+type KubeconfigReader struct {
+	provider providers.Provider
```

Comment:
Do we actually need to depend on the provider?
`UpdateKubeconfig` is a simple method
Can we just move/copy it here

If I'm not mistaken these changes you propose here will render UpdateKubeconfig in the provider useless. That's actually a great thing bc we can just remove that method from all providers

Reply:
We don't need the entire provider. I changed it to depend on docker's `ProviderClient` instead.